modules/cert-manager-override-arguments.adoc

```diff
@@ -0,0 +1,94 @@
+
+:_content-type: PROCEDURE
+[id="cert-manager-override-arguments_{context}"]
+= Customizing cert-manager by overriding arguments from the cert-manager Operator API
+
+You can override the default arguments for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig: <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=<host>:<port>' <2>
+      - '--dns01-recursive-nameservers-only' <3>
+      - '--acme-http01-solver-nameservers=<host>:<port>' <4>
+      - '--v=<verbosity_level>' <5>
+      - '--metrics-listen-address=<host>:<port>' <7>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check. For example, `--dns01-recursive-nameservers=1.1.1.1:53`.
+<3> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+<4> Provide a comma-separated list of `<host>:<port>` nameservers to query for the ACME HTTP01 self check. For example, `--acme-http01-solver-nameservers=1.1.1.1:53`.
+<5> Specify to set the log level verbosity to determine the amount of detail in logged error messages.
+<6> Specify the host and port for the metrics endpoint. The default value is `--metrics-listen-address=0.0.0.0:9402`.
+
+. Save the file to apply the changes.
+
+.Verification
+
+* Verify that the cert-manager pods are running by entering the following command:
++
+[source,terminal]
+----
+$ oc get pods -n cert-manager -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+name: cert-manager-6d4b5d4c97-kldwl
+namespace: cert-manager
+...
+  spec:
+    containers:
+    - args:
+      - --v=2
+      - --cluster-resource-namespace=$(POD_NAMESPACE)
+      - --leader-election-namespace=kube-system
+...
+name: cert-manager-cainjector-866c4fd758-ltxxj
+namespace: cert-manager
+...
+  spec:
+    containers:
+    - args:
+      - --v=2
+      - --leader-election-namespace=kube-system
+...
+name: cert-manager-webhook-6d48f88495-c88gd
+namespace: cert-manager
+...
+  spec:
+    containers:
+    - args:
+      - --v=2
+      - --secure-port=10250
+      - --dynamic-serving-ca-secret-namespace=$(POD_NAMESPACE)
+      - --dynamic-serving-ca-secret-name=cert-manager-webhook-ca
+      - --dynamic-serving-dns-names=cert-manager-webhook,cert-manager-webhook.$(POD_NAMESPACE),cert-manager-webhook.$(POD_NAMESPACE).svc
```

Comment:
`oc get -o yaml` will print fields alphabetically.
And the overridden arguments will be shown along with existing ones.
```suggestion
$ oc get pod <redeployed_cert-manager_pod> -n cert-manager -o yaml
----
+
.Example output. The overridden arguments will be added under the `env` field along with existing ones.
[source,yaml]
----
spec:
  containers:
  - args:
    - --acme-http01-solver-nameservers=1.1.1.1:53
    - --cluster-resource-namespace=$(POD_NAMESPACE)
    - --dns01-recursive-nameservers=1.1.1.1:53
    - --dns01-recursive-nameservers-only
    - --leader-election-namespace=kube-system
    - --max-concurrent-challenges=60
    - --metrics-listen-address=<host>:<port>
    - --v=6
```

Reply:
You didn't document `webhookConfig` and `cainjectorConfig` for other two components `cert-manager-cainjector` and `cert-manager-webhook`, so you need not display cert-manager-cainjector-xxxx and cert-manager-webhook-xxxx pods. But when I later review https://github.com/openshift/openshift-docs/pull/59777/files , I see it mentions `To use different log levels for different cert-manager components, see Customizing cert-manager by using the cert-manager Operator API fields`, but `Customizing cert-manager by using the cert-manager Operator API fields` does not document the how-to use different log levels for other two components `cert-manager-cainjector` and `cert-manager-webhook`. So you need to document for them by adding below in `oc edit certmanager cluster` part:
```
  webhookConfig:
    overrideArgs:
      - '--v=6'
  cainjectorConfig:
    overrideArgs:
      - '--v=6'
```